### PR TITLE
Replace ArrayList with List<T> in TableLayoutPanelBehavior

### PR DESCRIPTION
**Pull request by halgab**
Contributes to #8140

I fixed the field name in the same PR (in a separate commit) to make it easier to review all the call sites of this field

## Proposed changes

- Replace `ArrayList` `styles` with `List<TableLayoutStyle>` `_styles` in TableLayoutPanelBehavior

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10260)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/TableLayoutPanelBehavior.cs [hunk 1]
```diff
@@ -192,8 +192,8 @@ private bool CanResizeStyle(int[] widths)
     {
         bool canStealFromRight = false;
         bool canStealFromLeft = false;
-        int moveIndex = ((IList)styles).IndexOf(tableGlyph.Style);
-        if (moveIndex > -1 && moveIndex != styles.Count)
+        int moveIndex = ((IList)_styles).IndexOf(tableGlyph.Style);
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/TableLayoutPanelBehavior.cs:195]: Is this cast still needed?

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/TableLayoutPanelBehavior.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ internal class TableLayoutPanelBehavior : Behavior
     private TableLayoutPanel table;
     private StyleHelper rightStyle;
     private StyleHelper leftStyle;
-    private ArrayList styles; //List of the styles
+    private List<TableLayoutStyle> _styles; //List of the styles
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/TableLayoutPanelBehavior.cs:28]: Let's remove this comment. It doesn't seem to add much value here.

**Review by lonitra — COMMENTED** [review]
Some small comments otherwise LGTM!

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/TableLayoutPanelBehavior.cs (hunk 1 above)
Comment (halgab) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/TableLayoutPanelBehavior.cs:195]: I'm not sure it ever was needed...

**Review by lonitra — APPROVED** [review]

**Pull request merged**